File: Bit Manipulation/Python/clearBit.py
num = 5
print(f"Original Number: {num:b}") # Shift 1 to Nth bit then invert and AND with NUM;
def clearBit(num: int, bit:int) -> int:
    mask = ~(1 << bit)
    return num&mask


num = clearBit(num,0)
print(f"Clear bit: {num:b}")
      

File: Bit Manipulation/C/ClearBit.c
#include <stdio.h>
#include <stdlib.h>


int ClearBit(int n, int pos)
{
    int mask = ~(1 << pos);// create the pattern with a mask. i.e. 000010000. Inverted: 111101111
    return n & mask;
}

int main()
{
    int num = 21;

    printf("%d", ClearBit(num, 2));
    
    return 0;
}